src/odemis/acq/stitching/_registrar.py

```diff
@@ -14,18 +14,442 @@
+        self.tiles.append(tile)
+        if dependent_tiles != None:
+            self.num_dep_tiles = len(dependent_tiles)
+        
+    def getPositions(self):
+        """ 
+        returns:
+        tile_positions (list of tuples): the adjusted position in X/Y for each tile, in the order they were added
+        dep_tile_positions (list of tuples of tuples): the adjusted position for each dependent tile 
+        (in the order they were passed)
+        """ 
+        
+        positions = []
+        for i in range(len(self.tiles)):
+            positions.append(self.tiles[i].metadata[model.MD_POS])
+            
+        # Return same positions for dependent tiles
+        dep_tile_positions = []
+        for i in range(self.num_dep_tiles):
+            dep_tile_positions.append(positions)
+        
+        return positions, dep_tile_positions
+        
+        
+class ShiftRegistrar(object):
+    """ 
+    Locates the position of the image relative to the previous top tiles (horizontally and vertically) 
+    by using cross-correlation. The cross-correlation is done using just the part of the images which are 
+    supposed to be overlapping. In case the cross-correlation doesn't work (based on a couple of simple tests), 
+    fallback to the average shift on the same axis. 
+    """
+    
+    def __init__(self):
+        # arrays to store the vertical/horizontal shift values measured for each tile
+        self.nx = 1 # initialize grid to 1x1. The size will increase as new tiles are added.
+        self.ny = 1
+        self.coords_ver = [[None for _ in range(self.nx)] for _ in range(self.ny)] 
+        self.coords_hor = [[None for _ in range(self.nx)] for _ in range(self.ny)]
+        self.shifts = [[(0, 0) for _ in range(self.nx)] for _ in range(self.ny)]
+        self.tiles = [[None for _ in range(self.nx)] for _ in range(self.ny)]
+        self.last_ver = (0,0)  # last vertical shift measured
+        self.last_hor = (0,0)  # last horizontal shift measured
+        
+        # Initialize overlap. This will be modified after the second tile has been added.
+        # A value is needed to avoid errors when calculating the position for the first tile.
+        self.ovrlp = 0
+        
+        # List of 2D indices for grid positions in order of acquisition
+        self.acqOrder = []
+        
+        self.num_dep_tiles = 0 # required in getPositions() function
+
+       
+    def addTile(self,tile,dependent_tiles = None):
+        """ 
+        Extends grid by one tile. 
+        tile (DataArray of shape YX): each image must have at least MD_POS and MD_PIXEL_SIZE metadata. 
+        They should all have the same dtype.
+        dependent_tiles (list of DataArray or None): each of the dependent tile, where their position 
+        can be considered fixed relative to the main tile. Their content and metadata are not used
+        for the computation of the final position.
+        """  
+        
+        self.size = tile.shape[0]
```

Comment:
Check the tile shape is the same as the other tiles (=> just need to compare to the first tile). If not, raise ValueError.